KEYCLOAK-10849 KeycloakRole equals only with itself

Valeran86 opened the pull request:
I use Keycloak Spring Adapter (KSA) to secure existing application. Today I realized that some functions didn't work anymore because of security checking like this:
```
GrantedAuthority adminRole = new MySpecialGrantedAuthority( "superadmin" );
for ( GrantedAuthority role : userRoles ) {
        if ( role.equals( adminRole ) ) {
          return true;
        }
      }
```
In this example, when I use KSA authorization fails.
I believe, that more preferable in `KeycloakRole` use this implementation of `equals` method.

Previous discussion:
https://github.com/keycloak/keycloak/pull/6113

pedroigor commented:
@Valeran86, could you just change the commit message to `[KEYCLOAK-10849] - KeycloakRole equals only with itself`. After that, I'll just merge :)

pedroigor reviewed (changes requested):
Missing reference to JIRA in commit message. After that we can merge.

Valeran86 commented:
@pedroigor, thanks for your reply. I apologize for long time of answer. Now I changed commit message as you recommended :)

abstractj commented:
@Valeran86 could you please rebase your PR with the changes from master? Seems like Travis is stable again.

Valeran86 committed:
[KEYCLOAK-10849] - KeycloakRole equals only with itself

I use Keycloak Spring Adapter (KSA) to secure existing application. Today I realized that some functions didn't work anymore because of security checking like this:
```
GrantedAuthority adminRole = new MySpecialGrantedAuthority( "superadmin" );
for ( GrantedAuthority role : userRoles ) {
        if ( role.equals( adminRole ) ) {
          return true;
        }
      }
```
In this example, when I use KSA authorization fails.
I believe, that more preferable in `KeycloakRole` use this implementation of `equals` method.

Valeran86 commented:
@abstractj , Hi! Yep, sure.
